### PR TITLE
Allow for specified Cargo commands to be run on a subset of the integration tests

### PR DESCRIPTION
**Pull request by david-perez**
This commit allows for the `modules` and `cargoCommands` properties to
be specified when running the `codegen-test` and `codegen-server-test`
Gradle modules. The `modules` property allows one to only generate a
subset of the integration test services, while the `cargoCommands`
property allows one to specify the Cargo commands to be run on the
generated Rust crates.

This functionality can be useful to reduce development iteration cycles. For
instance, to only run `cargo test` on the integration test
`simple.smithy`:

```sh
./gradlew codegen-test:build -P cargoCommands='test' -P modules='simple'
```

This commit also refactors the buildscripts of the `codegen-test` and
the `codegen-server-test` modules, extracting shared functionality to
`buildSrc/src/main/kotlin/CodegenTestCommon.kt`.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: codegen-test/README.md [hunk 1]
```diff
@@ -1,8 +1,6 @@
...
 
-The `test` task will run `cargo check` and `cargo clippy` to validate that the generated Rust compiles and is idiomatic.
-## Usage
-```
-../gradlew test
-```
+Refer to `../codegen-test/README.md` for documentation on how to use this
+module. This module is analogous to the `codegen-test` one, but it's named
+`codegen-server-test` and it runs the `rust-server-codegen` Smithy build
+plugin instead.
```
[the remark this answers is not in the document]
Comment (david-perez) [reply codegen-test/README.md]: Indeed, I swapped the `README`s' contents.

**Review by crisidev — APPROVED** [review]

File: buildSrc/src/main/kotlin/CodegenTestCommon.kt [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    CLIPPY("cargoClippy");
+}
+
+fun generateCargoWorkspace(pluginName: String, tests: List<CodegenTest>) =
+    """
+    [workspace]
+    members = [
+        ${tests.joinToString(",") { "\"${it.module}/$pluginName\"" }}
+    ]
+    """
```
[the remark this answers is not in the document]
Comment (rcoh) [reply buildSrc/src/main/kotlin/CodegenTestCommon.kt:58]: because this block doesn't have trimIndent

[the remark this answers is not in the document]
Comment (david-perez) [reply buildSrc/src/main/kotlin/CodegenTestCommon.kt]: Fixed https://github.com/awslabs/smithy-rs/pull/1165/commits/4773fbd877e8cf803ff6b95957d47edf142ff786

File: buildSrc/src/main/kotlin/CodegenTestCommon.kt [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+ * The list of Cargo commands that is run by default is defined in [AllCargoCommands].
+ */
+fun cargoCommands(properties: PropertyRetriever): List<Cargo> {
+    val cargoCommandsOverride = properties.get("cargoCommands")?.split(",")?.map { it.trim() }?.map {
+        when(it) {
+            "check" -> Cargo.CHECK
+            "test" -> Cargo.TEST
+            "docs" -> Cargo.DOCS
+            "clippy" -> Cargo.CLIPPY
+            else -> throw InvalidParameterException("Unexpected Cargo command `$it` (valid commands are `check`, `test`, `docs`, `clippy`)")
```
Comment (jdisanti) [inline buildSrc/src/main/kotlin/CodegenTestCommon.kt:90]: Nit: `IllegalArgumentException` is probably a better exception type to use here. This one is for JCA/JCE engine classes.

**Review by jdisanti — APPROVED** [review]
This is awesome!

**Pull request merged**